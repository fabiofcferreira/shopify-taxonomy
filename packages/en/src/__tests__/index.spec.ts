import { TaxonomyLanguage } from "shopify-taxonomy-core";
import client from "..";

describe("English taxonomy client", () => {
  it("language should be set to English", () => {
    expect(client.getLanguage()).toBe(TaxonomyLanguage.English);
  });

  it("should return English taxonomy categories", () => {
    expect(client.getCategory("ap-2-1-1")).toMatchSnapshot();
    expect(client.getCategory("hb-1-2")).toMatchSnapshot();
  });

  it("should return null for not found categories ", () => {
    expect(client.getCategory("ap-12313123")).toBe(null);
    expect(client.getCategory("hb-100")).toBe(null);
    expect(client.getCategory("ww")).toBe(null);
  });

  it("should throw if category is invalid", () => {
    expect(() => client.getCategory("hb---11")).toThrowErrorMatchingSnapshot();
    expect(() => client.getCategory("h")).toThrowErrorMatchingSnapshot();
    expect(() => client.getCategory("a4")).toThrowErrorMatchingSnapshot();
  });
});
