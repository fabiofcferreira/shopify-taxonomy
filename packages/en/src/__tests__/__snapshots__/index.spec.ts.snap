// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`English taxonomy client should return English taxonomy categories 1`] = `
{
  "fullName": "Animals & Pet Supplies > Pet Supplies > Bird Supplies > Bird Cage Accessories",
  "id": "ap-2-1-1",
  "name": "Bird Cage Accessories",
  "parent_id": "ap-2-1",
}
`;

exports[`English taxonomy client should return English taxonomy categories 2`] = `
{
  "fullName": "Health & Beauty > Health Care > Bed Pans",
  "id": "hb-1-2",
  "name": "Bed Pans",
  "parent_id": "hb-1",
}
`;

exports[`English taxonomy client should throw if category is invalid 1`] = `"Category ID 'hb---11' is not valid."`;

exports[`English taxonomy client should throw if category is invalid 2`] = `"Category ID 'h' is not valid."`;

exports[`English taxonomy client should throw if category is invalid 3`] = `"Category ID 'a4' is not valid."`;
